packages/lodestar/src/db/api/beacon/repositories/attestations.ts

```diff
@@ -12,4 +12,20 @@ export class AttestationRepository extends BulkRepository<Attestation> {
     db: IDatabaseController) {
     super(config, db, Bucket.attestation, config.types.Attestation);
   }
+
+  public async setUnderRoot(attestation: Attestation): Promise<void> {
+    return super.set(hashTreeRoot(attestation.data, this.config.types.AttestationData), attestation);
+  }
+
+  public async deleteManyByValue(values: Attestation[]): Promise<void> {
+    await this.deleteMany(values.map(value => hashTreeRoot(value.data, this.config.types.AttestationData)));
+  }
+
+  public async deleteAll(idFunction?: (value: Attestation) => Id): Promise<void> {
+    const data = await this.getAll();
+    const defaultIdFunction: (value: Attestation) => Id =
```

Comment:
what syntax is this ? Lol

Reply:
its typescript :D
anonymus function signature :)